source/gloperate-glkernel/CMakeLists.txt

```diff
@@ -0,0 +1,208 @@
+find_package(globjects REQUIRED)
+find_package(glkernel  REQUIRED)
+find_package(cppexpose REQUIRED)
+find_package(cpplocate REQUIRED)
+find_package(cppassist REQUIRED)
+
+#
+# Library name and options
+#
+
+# Target name
+set(target gloperate-glkernel)
+
+# Set API export file and macro
+string(TOUPPER ${target} target_upper)
+set(export_file  "include/${target}/${target}_api.h")
+set(export_macro "${target_upper}_API")
+
+
+#
+# Sources
+#
+
+set(include_path "${CMAKE_CURRENT_SOURCE_DIR}/include/${target}")
+set(source_path  "${CMAKE_CURRENT_SOURCE_DIR}/source")
+
+set(headers
+    ${include_path}/DiscDistributionKernelStage.h
+    ${include_path}/HemisphereDistributionKernelStage.h
+    ${include_path}/NoiseKernelStage.h
+    ${include_path}/TransparencyKernelStage.h
+)
+
+set(sources
+    ${source_path}/DiscDistributionKernelStage.cpp
+    ${source_path}/HemisphereDistributionKernelStage.cpp
+    ${source_path}/NoiseKernelStage.cpp
+    ${source_path}/TransparencyKernelStage.cpp
+)
+
+# Group source files
+set(header_group "Header Files (API)")
+set(source_group "Source Files")
+source_group_by_path(${include_path} "\\\\.h$|\\\\.hpp$"
+    ${header_group} ${headers})
+source_group_by_path(${source_path}  "\\\\.cpp$|\\\\.c$|\\\\.h$|\\\\.hpp$"
+    ${source_group} ${sources})
+
+
+#
+# Create library
+#
+
+# Build library
+add_library(${target}
+    ${sources}
+    ${headers}
+)
+
+# Create namespaced alias
+add_library(${META_PROJECT_NAME}::${target} ALIAS ${target})
+
+# Export library for downstream projects
+export(TARGETS ${target} NAMESPACE ${META_PROJECT_NAME}:: FILE ${PROJECT_BINARY_DIR}/cmake/${target}/${target}-export.cmake)
+
+# Create API export header
+generate_export_header(${target}
+    EXPORT_FILE_NAME  ${export_file}
+    EXPORT_MACRO_NAME ${export_macro}
+)
+
+
+#
+# Project options
+#
+
+set_target_properties(${target}
+    PROPERTIES
+    ${DEFAULT_PROJECT_OPTIONS}
+    FOLDER "${IDE_FOLDER}"
+)
+
+
+#
+# Include directories
+#
+
+target_include_directories(${target}
+    PRIVATE
+    ${DEFAULT_INCLUDE_DIRECTORIES}
+    ${PROJECT_BINARY_DIR}/source/include
+    ${CMAKE_CURRENT_SOURCE_DIR}/include
+    ${CMAKE_CURRENT_BINARY_DIR}/include
+    ${PROJECT_SOURCE_DIR}/source/gloperate/include
+    ${GLM_INCLUDE_DIR}
+
+    PUBLIC
+
+    INTERFACE
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/include>
+    $<INSTALL_INTERFACE:include>
+)
+
+
+#
+# Libraries
+#
+
+target_link_libraries(${target}
+    PRIVATE
+    cppassist::cppassist
+    cppexpose::cppexpose
+    cpplocate::cpplocate
+    glkernel::glkernel
+    globjects::globjects
+    glbinding::glbinding
+    libzeug::signalzeug
+    ${META_PROJECT_NAME}::gloperate
+
+    PUBLIC
+    ${DEFAULT_LIBRARIES}
+
+    INTERFACE
+)
+
+
+#
+# Compile definitions
+#
+
+target_compile_definitions(${target}
+    PRIVATE
+
+    PUBLIC
+    $<$<NOT:$<BOOL:${BUILD_SHARED_LIBS}>>:${target_upper}_STATIC_DEFINE>
```

Comment:
We use ```${target_id}``` now, please refer to https://github.com/cginternals/cmake-init/blob/master/source/baselib/CMakeLists.txt#L150.